components/tidb_query_datatype/src/codec/mysql/time/mod.rs

```diff
@@ -779,7 +779,18 @@ impl Time {
         fsp: i8,
         round: bool,
     ) -> Result<Time> {
-        parser::parse_from_decimal(ctx, input, time_type, check_fsp(fsp)?, round)
+        parser::parse_from_float_string(ctx, input.to_string(), time_type, check_fsp(fsp)?, round)
```

Comment:
It seems weird that `parse_from_decimal` calls `parse_from_float_string`.

Reply:
It is used by parse_from_decimal and parse_from_real, decimal and real will convert to string (float format).